src/main/java/Project.java

```diff
@@ -0,0 +1,142 @@
+import java.util.ArrayList;
+
+public class Project {
+
+    private Professor projectProf;
+    private ArrayList<Student> students;
+
+    private String description;
+    private ArrayList<String> restrictions;
+    private int maxCapacity;
+    private int currentCapacity;
+
+    private boolean isArchived;
+    private boolean isDeleted;
+
+
+    public Project(Professor projectProf, String description, ArrayList<String> restrictions, int maxCapacity)
+    {
+        this.projectProf = projectProf;
+        this.students = new ArrayList<>();
+
+        this.description = description;
+        this.restrictions = restrictions;
+        this.maxCapacity = maxCapacity;
+        this.currentCapacity = students.size();
+
+        isArchived = false;
+        isDeleted = false;
+    }
+
+
+    /**
+     * Attempt to add a student to the project. Check against capacity & restrictions
+     * @param student Student to be added
+     * @return Boolean whether or not the student was added
+     */
+    public boolean addStudent(Student student)
+    {
+        if (currentCapacity < maxCapacity)
```

Comment:
Put a check to ensure the project is not archived

Reply:
Fixed 👍🏼